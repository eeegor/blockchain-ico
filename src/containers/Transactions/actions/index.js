import {
	getTransactionsActions,
	getTransactions,
	getTransactionsRequest,
	getTransactionsSuccess,
	getTransactionsError
} from './getTransactions';
import {
	setTransactionStageActions,
	setTransactionStage
} from './setTransactionStage';

export const actions = {
	...getTransactionsActions,
	...setTransactionStageActions
};

export {
	getTransactions,
	getTransactionsRequest,
	getTransactionsSuccess,
	getTransactionsError,
	setTransactionStage
};
