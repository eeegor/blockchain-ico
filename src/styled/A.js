import styled from 'styled-components';
import { darken } from 'polished';
import { theme } from './theme';
import { helperMargin } from './helperMargin';

export const A = styled.a`
	font-size: 16px;
	letter-spacing: 0.5px;
	line-height: 26px;
	font-family: ${theme.fonts.gtCinetype};
	color: ${theme.colors.buttonPrimary};
	text-decoration: none;
	transition: all 0.15s ease-out;

	:hover {
		color: ${darken(0.2, theme.colors.buttonPrimary)};
	}

	${props =>
		props.color &&
		`
	  color: ${props.color} !important;
  `};

	${props => helperMargin(props)}
`;
