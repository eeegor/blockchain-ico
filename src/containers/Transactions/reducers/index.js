import { actions } from '../actions';

/**
 *
 * @param {object} state
 * @param {object} action
 * @returns {object}
 */

export const transactionReducer = (state = {}, action) => {
	// console.log(action.type, action);

	const { payload } = action;

	switch (action.type) {
		case actions.GET_TRANSACTIONS_REQUEST:
			return {
				...state
			};

		case actions.GET_TRANSACTIONS_SUCCESS:
			return {
				...state,
				data: payload.data
			};

		case actions.GET_TRANSACTIONS_ERROR:
			return {
				...state,
				data: []
			};

		case actions.SET_TRANSACTIONS_STAGE:
			return {
				...state,
				stage: payload.stage || 'all'
			};

		default:
			return state;
	}
};
