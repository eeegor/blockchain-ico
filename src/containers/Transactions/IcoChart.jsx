import React from 'react';
import moment from 'moment';
import {
	Chart,
	ChartInfo,
	TextSmall,
	TitleUppercase,
	Column,
	Tooltip
} from '../../styled';
import { theme } from '../../styled/theme';
import { XYFrame } from 'semiotic';
import { windowMaxWidth } from '../../util';

const selected = {
	all: {
		dateMin: '2017-07-15T10:00:00Z',
		dateMax: '2018-06-10T10:10:00Z',
		totalMin: 0,
		totalMax: 2000000,
		scale: 1500
	},
	preIco: {
		dateMin: '2017-08-02T03:00:00Z',
		dateMax: '2017-08-07T12:00:00Z',
		totalMin: 0,
		totalMax: 1000000,
		scale: 750
	},
	ico: {
		dateMin: '2017-10-31T06:00:00Z',
		dateMax: '2017-11-21T18:45:00Z',
		totalMin: 780000,
		totalMax: 1250000,
		scale: 500
	},
	finalIco: {
		dateMin: '2018-05-09T02:00:00Z',
		dateMax: '2018-05-17T10:00:00Z',
		totalMin: 1110000,
		totalMax: 1800000,
		scale: 750
	}
};

const tooltipCreator = hover => {
	const isTablet = windowMaxWidth() > theme.media.tablet;
	return (
		<Tooltip className="tooltip-content">
			<Column>
				<TextSmall>Index</TextSmall>
				<TitleUppercase color={theme.colors.primaryGreen}>
					{hover.order}
				</TitleUppercase>
			</Column>
			<Column>
				<TextSmall>USD / {hover.currency}:</TextSmall>
				<TitleUppercase>
					${hover.usd.toFixed(2)} / {hover.normalizedValue}
				</TitleUppercase>
			</Column>
			{isTablet && (
				<>
					<Column>
						<TextSmall>Received:</TextSmall>
						<TitleUppercase>
							{moment(hover.received).format('DD.MM.YY, hh:mm')}
						</TitleUppercase>
					</Column>
					<Column>
						<TextSmall>Confirmed:</TextSmall>
						<TitleUppercase>
							{moment(hover.confirmed).format('DD.MM.YY, hh:mm')}
						</TitleUppercase>
					</Column>
				</>
			)}
			<Column>
				<TextSmall>Total:</TextSmall>
				<TitleUppercase color={theme.colors.primaryGreen}>
					$
					{hover.sum.toLocaleString({
						minimumFractionDigits: 2,
						maximumFractionDigits: 2
					})}
				</TitleUppercase>
			</Column>
		</Tooltip>
	);
};

export const IcoChart = props => {
	const { data, stage } = props;
	const isMobile = windowMaxWidth() < theme.media.tablet;
	const frameHeight = isMobile ? 400 : 600;
	const frameWidth =
		windowMaxWidth() < 1030 - 24 ? windowMaxWidth() - 24 : 1030 - 24;

	if (!stage) {
		return;
	}

	return (
		<Chart maxWidth={frameWidth} noPadding>
			<ChartInfo>
				<TextSmall>
					{isMobile ? 'Tap on' : 'Hover'} the chart to receive
					transaction info
				</TextSmall>
			</ChartInfo>
			<XYFrame
				size={[frameWidth, frameHeight]}
				lines={data}
				points={data}
				xAccessor={'received'}
				yAccessor={'sum'}
				xExtent={[
					new Date(selected[stage].dateMin),
					new Date(selected[stage].dateMax)
				]}
				yExtent={[selected[stage].totalMin, selected[stage].totalMax]}
				lineStyle={{
					stroke: theme.colors.primaryCyan,
					strokeWidth: 2,
					strokeOpacity: 0.3
				}}
				tooltipContent={hover => tooltipCreator(hover)}
				hoverAnnotation={true}
				pointStyle={d => {
					let fill;
					switch (d.currency) {
						case 'ETH':
							fill = theme.colors.primaryGreen;
							break;
						case 'BTC':
							fill = theme.colors.primaryRed;
							break;
						case 'LTC':
							fill = theme.colors.primaryPurple;
							break;
						default:
							fill = theme.colors.primaryLightBlueV2;
							break;
					}
					return {
						fill,
						r: 3,
						strokeWidth: 1,
						fillOpacity: 0.75
					};
				}}
				axes={[
					{
						orient: 'left',
						tickFormat: d => `$${d.toLocaleString()}`,
						ticks: isMobile ? 3 : 4,
						rotate: -90
					},
					{
						orient: 'bottom',
						tickFormat: d => (
							<g
								className="axis-label axis-label--date"
								textAnchor="top"
							>
								<text
									className="axis-label__date"
									x={-32}
									y={10}
								>
									{moment(d).format('DD.MM.YYYY')}
								</text>
								<text
									className="axis-label__time"
									x={-32}
									y={24}
								>
									{moment(d).format('hh:mm')}
								</text>
							</g>
						),
						ticks: isMobile ? 3 : 8
					}
				]}
				margin={
					isMobile
						? { top: 72, left: 20, bottom: 72, right: 8 }
						: { top: 96, left: 32, bottom: 96, right: 16 }
				}
			/>
		</Chart>
	);
};
