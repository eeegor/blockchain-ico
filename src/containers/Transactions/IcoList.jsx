import React from 'react';
import moment from 'moment';
import { formatValue, currencyColor } from '../../util';
import {
	Div,
	Container,
	Bar,
	BarInfo,
	TextSmall,
	BarControls,
	Select,
	Button,
	Grid,
	CardTransaction,
	CardIcon,
	CardBody,
	TitleUppercase
} from '../../styled';
import { IconBitcoin, IconLitecoin, IconEtherium } from '../../components';
import { theme } from '../../styled/theme';

export const IcoList = props => {
	const {
		filteredTransactions,
		prices,
		page,
		perPage,
		totalPages,
		handlePerPage,
		handlePrevPage,
		handleNextPage
	} = props;

	if (!filteredTransactions) {
		return null;
	}
	return (
		<Div>
			<Container>
				<Bar centerY justifyBetween>
					<BarInfo paddingY="24" centerY>
						<TextSmall marginBottom="0" marginRight="12">
							{filteredTransactions.length} Transactions
						</TextSmall>
						<TextSmall marginBottom="0" marginRight="12">
							Page {page} / {totalPages}
						</TextSmall>
					</BarInfo>
					<BarControls>
						<Bar centerY>
							<TextSmall marginBottom="0" marginRight="6">
								Per page
							</TextSmall>
							<Select
								className="per-page"
								value={perPage}
								onChange={e =>
									handlePerPage(
										e.target.value,
										filteredTransactions.length
									)
								}
							>
								<select>
									<option value="9">9</option>
									<option value="30">30</option>
									<option value="60">60</option>
									<option value="99">99</option>
								</select>
							</Select>
						</Bar>
						<Bar>
							<Button
								className="prev-page"
								noShadow
								marginX="12"
								paddingX="12"
								onClick={() => handlePrevPage()}
							>
								Prev
							</Button>
							<Button
								paddingX="12"
								className="next-page"
								noShadow
								onClick={() => handleNextPage(totalPages)}
							>
								Next
							</Button>
						</Bar>
					</BarControls>
				</Bar>
			</Container>
			<Div background={theme.colors.darkBg} paddingY="24">
				<Container>
					<Grid className="transactions-list" columns={[1, 2, 3]}>
						{filteredTransactions
							.slice(perPage * (page - 1), perPage * page)
							.map(item => (
								<CardTransaction
									key={item.txid}
									withIcon
									noBg
									padding="0"
								>
									<CardIcon>
										{item.currency === 'BTC' && (
											<IconBitcoin />
										)}
										{item.currency === 'LTC' && (
											<IconLitecoin />
										)}
										{item.currency === 'ETH' && (
											<IconEtherium />
										)}
									</CardIcon>
									<CardBody>
										<Bar
											marginBottom={12}
											className="transaction__currency-usd"
										>
											<TitleUppercase
												color={currencyColor(
													item.currency
												)}
												className="transaction__currency"
												marginBottom="0"
												marginRight="12"
											>
												{item.currency}:
											</TitleUppercase>
											<TitleUppercase
												className="transaction__usd"
												color={
													theme.colors.primaryGreen
												}
											>
												$
												{formatValue(
													item.value,
													item.currency,
													prices,
													'USD'
												).toFixed(2)}
											</TitleUppercase>
										</Bar>
										<Bar className="transaction__index">
											<TextSmall marginRight="6">
												Transaction index:
											</TextSmall>
											<TextSmall color="#fff">
												{item.order}
											</TextSmall>
										</Bar>
										<TextSmall
											className="transaction__received"
											marginBottom="0"
											marginRight="12"
										>
											Received:{' '}
											{moment(item.received).format(
												'DD.MM.YY, HH:mm:ss'
											)}
										</TextSmall>
										<TextSmall
											className="transaction__confirmed"
											marginBottom="0"
											marginRight="12"
										>
											Confirmed:{' '}
											{moment(item.confirmed).format(
												'DD.MM.YY, HH:mm:ss'
											)}
										</TextSmall>
									</CardBody>
								</CardTransaction>
							))}
					</Grid>
				</Container>
			</Div>
		</Div>
	);
};
