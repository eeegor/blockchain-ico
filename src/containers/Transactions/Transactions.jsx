import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { IconBitcoin, IconEtherium, IconLitecoin } from '../../components';
import {
	Container,
	H2,
	Grid,
	Card,
	Button,
	CardIcon,
	TitleUppercase,
	Text,
	TextSmall,
	Bar,
	Hr
} from '../../styled';
import { getTransactions, setTransactionStage } from './actions';
import { formatValue } from '../../util';

export const TransactionsContainer = styled.div`
	margin-top: 84px;
`;

const buttonData = [
	{
		label: 'All',
		selectStage: '',
		selected: stage => !stage || stage === 'all' || stage === ''
	},
	{
		label: 'Pre Ico',
		selectStage: 'preIco',
		selected: stage => stage === 'preIco'
	},
	{ label: 'Ico', selectStage: 'ico', selected: stage => stage === 'ico' },
	{
		label: 'Final Ico',
		selectStage: 'finalIco',
		selected: stage => stage === 'finalIco'
	}
];

export class Transactions extends Component {
	state = {
		page: 2,
		perPage: 30
	};

	componentDidMount() {
		const { getTransactions } = this.props;
		getTransactions && getTransactions();
	}

	selectStage = (event, stage = '') => {
		event && event.preventDefault();
		const { setTransactionStage } = this.props;
		setTransactionStage && setTransactionStage({ stage });
		this.gotoPage(1);
	};

	gotoPage = page => this.setState(state => ({ ...state, page }));

	handlePerPage = (nextPerPage, totalItems) =>
		this.setState(state => {
			const nextMaxPages = Math.ceil(totalItems / nextPerPage);
			return {
				...state,
				page: state.page < nextMaxPages ? state.page : nextMaxPages,
				perPage: nextPerPage
			};
		});

	handlePrevPage = () =>
		this.setState(state => ({
			...state,
			page: state.page > 1 ? state.page - 1 : 1
		}));

	handleNextPage = totalPages =>
		this.setState(state => ({
			...state,
			page: state.page + 1 < totalPages ? state.page + 1 : totalPages
		}));

	render() {
		const { page, perPage } = this.state;
		const { transactions, stage } = this.props;
		const filterTransactions = (transactions, stage) => {
			if (!transactions) {
				return [];
			}
			if (!stage || stage === 'all') {
				return transactions;
			}
			return transactions.filter(item => item.stage === stage);
		};
		const filteredTransactions = filterTransactions(transactions, stage);
		const totalPages =
			filteredTransactions &&
			Math.ceil(filteredTransactions.length / perPage);

		return (
			<TransactionsContainer>
				<Container>
					<H2 marginTop="48">Transactions by stage</H2>
					<Grid gap="6" columns={[4, 7, 8]} marginY="24">
						{buttonData.map((button, index) => (
							<Button
								paddingX="0"
								noShadow
								key={index}
								onClick={e =>
									this.selectStage(e, button.selectStage)
								}
								selected={button.selected(stage)}
							>
								{button.label}
							</Button>
						))}
					</Grid>
					<Bar marginTop="24" centerY justifyBetween>
						<Bar paddingY="24" centerY>
							<Text marginBottom="0" marginRight="12">
								{filteredTransactions.length} Transactions
							</Text>
							<Text marginBottom="0" marginRight="12">
								Page {page} / {totalPages}
							</Text>
						</Bar>
						<Bar>
							<select
								className="per-page"
								value={perPage}
								onChange={e =>
									this.handlePerPage(
										e.target.value,
										filteredTransactions.length
									)
								}
							>
								<option value="">Per page</option>
								<option value="9">9</option>
								<option value="30">30</option>
								<option value="60">60</option>
								<option value="99">99</option>
							</select>
							<Button
								className="prev-page"
								noShadow
								marginX="12"
								paddingX="12"
								onClick={() => this.handlePrevPage()}
							>
								Prev
							</Button>
							<Button
								paddingX="12"
								className="next-page"
								noShadow
								onClick={() => this.handleNextPage(totalPages)}
							>
								Next
							</Button>
						</Bar>
					</Bar>
					<Hr />
					<Grid className="transactions-list" columns={3}>
						{filteredTransactions
							.slice(perPage * (page - 1), perPage * page)
							.map(item => (
								<Card key={item.txid} withIcon noBg padding="0">
									<CardIcon>
										{item.currency === 'BTC' && (
											<IconBitcoin />
										)}
										{item.currency === 'LTC' && (
											<IconLitecoin />
										)}
										{item.currency === 'ETH' && (
											<IconEtherium />
										)}
									</CardIcon>
									<Container center>
										<TitleUppercase
											marginBottom="0"
											marginRight="12"
										>
											{item.currency}
										</TitleUppercase>
										<Text marginBottom="0" marginRight="12">
											{item.stage}
										</Text>
										<TextSmall
											marginBottom="0"
											marginRight="12"
										>
											$
											{formatValue(
												item.value,
												item.currency,
												'USD'
											).toFixed(2)}
										</TextSmall>
									</Container>
								</Card>
							))}
					</Grid>
				</Container>
			</TransactionsContainer>
		);
	}
}

const mapStateToProps = state => {
	console.log('mapStateToProps', state);
	return {
		app: state.app,
		transactions: state.transactions.data,
		stage: state.transactions.stage
	};
};

export default connect(
	mapStateToProps,
	{ getTransactions, setTransactionStage }
)(Transactions);
