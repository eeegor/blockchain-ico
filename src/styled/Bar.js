import styled from 'styled-components';
import { helperMargin } from './helperMargin';
import { helperPadding } from './helperPadding';
import { theme } from './theme';

export const Bar = styled.div`
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;

	${props =>
		props.center &&
		`
		align-items: center;
		justify-content: center;
	`}

	${props =>
		props.centerX &&
		`
		justify-content: center;
	`}
	
	${props =>
		props.justifyBetween &&
		`
		justify-content: space-between;
	`}
	
	${props =>
		props.centerY &&
		`
		align-items: center;
	`}

	${props => helperMargin(props)}
	${props => helperPadding(props)}
`;

export const BarInfo = styled(Bar)`
	@media (max-width: ${theme.media.tablet - 1}px) {
		width: 100%;
		justify-content: space-between;
	}
`;

export const BarControls = styled(Bar)`
	@media (max-width: ${theme.media.tablet - 1}px) {
		width: 100%;
		justify-content: space-between;
		margin-bottom: 12px;
	}
`;
