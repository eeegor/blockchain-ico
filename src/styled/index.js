import { H1 } from './H1';
import { H2 } from './H2';
import { H4 } from './H4';
import { H5 } from './H5';
import { H6 } from './H6';
import { Hr } from './Hr';
import { A } from './A';
import { Text } from './Text';
import { TextSmall } from './TextSmall';
import { TitleUppercase } from './TitleUppercase';
import { Paragraph } from './Paragraph';
import { Container } from './Container';
import { Grid } from './Grid';
import { Button } from './Button';
import { Header } from './Header';
import { Content } from './Content';
import { Card } from './Card';
import { CardIcon } from './CardIcon';

export {
	H1,
	H2,
	H4,
	H5,
	H6,
	Hr,
	A,
	Text,
	TextSmall,
	TitleUppercase,
	Paragraph,
	Container,
	Grid,
	Button,
	Header,
	Content,
	Card,
	CardIcon,
};
