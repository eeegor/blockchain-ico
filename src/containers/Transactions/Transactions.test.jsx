import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Transactions } from './Transactions';
import { mount } from 'enzyme';
import { CardTransaction } from '../../styled';

afterEach(() => {
	jest.clearAllMocks();
	jest.restoreAllMocks();
});

describe('Transactions', () => {
	const getTransactions = jest.fn();
	const transactions = [];
	const stage = 'all';
	const loading = false;
	const prices = {
		fetched: moment(new Date()).format('DD.MM.YY HH:mm:ss')
	};
	const props = {
		getTransactions,
		transactions,
		stage,
		loading,
		prices
	};

	it('renders Transactions', () => {
		const tree = renderer.create(<Transactions {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('switches stage', () => {
		const wrapper = mount(<Transactions {...props} />);
		setTimeout(() => {
			wrapper
				.find('button')
				.first()
				.simulate('click');
			expect(getTransactions).toHaveBeenCalledTimes(1);
		}, 1200);
	});

	it('shows prev transactions page', () => {
		const wrapper = mount(<Transactions {...props} />);
		setTimeout(() => {
			wrapper
				.find('.prev-page')
				.first()
				.simulate('click');
			expect(getTransactions).toHaveBeenCalledTimes(1);
			expect(wrapper.state().page).toEqual(1);
		}, 1200);
	});

	it('shows next transactions page', () => {
		const transactions = [
			{ txid: '212122112121', value: '1222000000000', currency: 'ETH' },
			{ txid: '242343432332', value: '393832323', currency: 'BTC' },
			{ txid: '992392383332', value: '32323232', currency: 'LTC' }
		];
		const state = {
			page: 1,
			perPage: 2
		};
		const wrapper = mount(
			<Transactions {...props} transactions={transactions} />
		);
		wrapper.setState(state);
		setTimeout(() => {
			wrapper
				.find('.next-page')
				.first()
				.simulate('click');
			expect(getTransactions).toHaveBeenCalledTimes(1);
			expect(wrapper.state().page).toEqual(2);
			expect(wrapper.state().perPage).toEqual(2);
		}, 1200);
	});

	it('handles items per page', () => {
		const transactions = [
			{ txid: '212122112121', value: '1222000000000', currency: 'ETH' },
			{ txid: '242343432332', value: '393832323', currency: 'BTC' },
			{ txid: '992392383332', value: '32323232', currency: 'LTC' }
		];
		const state = {
			page: 1,
			perPage: 9
		};
		const wrapper = mount(
			<Transactions {...props} transactions={transactions} />
		);
		wrapper.setState(state);
		setTimeout(() => {
			wrapper
				.find('.per-page')
				.first()
				.simulate('change', { target: { value: 99 } });
			expect(wrapper.state().perPage).toEqual(99);
		}, 1200);
	});

	it('filters transactions by stage', () => {
		const transactions = [
			{
				txid: '212122112121',
				value: '1222000000000',
				currency: 'ETH',
				stage: 'ico'
			},
			{
				txid: '723939839833',
				value: '44399993333',
				currency: 'ETH',
				stage: 'ico'
			},
			{
				txid: '242343432332',
				value: '393832323',
				currency: 'BTC',
				stage: 'preIco'
			},
			{
				txid: '992392383332',
				value: '32323232',
				currency: 'LTC',
				stage: 'finalIco'
			}
		];
		const state = {
			page: 1,
			perPage: 3
		};
		const wrapper = mount(
			<Transactions {...props} transactions={transactions} stage="ico" />
		);
		wrapper.setState(state);
		setTimeout(() => {
			const cards = wrapper
				.find('.transactions-list')
				.find(CardTransaction);
			expect(cards.length).toEqual(2);
		}, 1200);
	});
});
