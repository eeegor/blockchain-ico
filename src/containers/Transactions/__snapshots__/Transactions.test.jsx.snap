// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions renders Transactions 1`] = `
.c2 {
  font-size: 24px;
  line-height: 33px;
  font-family: Rubik-Regular;
  margin-bottom: 12px;
  margin-top: 48px !important;
}

.c12 {
  width: 100%;
  margin: 26px auto 28px auto;
  outline: 0;
  border: 0;
  border-top: 1px solid #3332;
  max-width: 1006px;
}

.c8 {
  font-size: 16px;
  line-height: 26px;
  font-family: GTCinetype-Regular;
  margin-bottom: 12px;
  margin-bottom: 0 !important;
  margin-right: 12px !important;
}

.c1 {
  padding: 0 12px;
  margin-left: auto;
  margin-right: auto;
  height: 100%;
  width: 100%;
  max-width: 1030px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 4,1fr );
  grid-gap: 6px 6px;
  margin-top: 24px !important;
  margin-bottom: 24px !important;
}

.c13 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 1,1fr );
}

.c4 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  box-shadow: none !important;
  color: #fff;
  background: #2B60FF;
  border: 1px solid #2B60FF;
  padding-left: 0 !important;
  padding-right: 0 !important;
}

.c4:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c5 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  box-shadow: none !important;
  padding-left: 0 !important;
  padding-right: 0 !important;
}

.c5:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c10 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  box-shadow: none !important;
  margin-left: 12px !important;
  margin-right: 12px !important;
  padding-left: 12px !important;
  padding-right: 12px !important;
}

.c10:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c11 {
  cursor: pointer;
  border: 1px solid #ebf0ff;
  outline: 0;
  background: #fff;
  color: #2B60FF;
  padding: 0.75rem 3rem;
  border-radius: 4px;
  font-size: 13px;
  -webkit-letter-spacing: 1.18px;
  -moz-letter-spacing: 1.18px;
  -ms-letter-spacing: 1.18px;
  letter-spacing: 1.18px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 0;
  font-family: GTCinetype-Bold;
  box-shadow: 0 14px 17px rgba(0,4,18,.16);
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  box-shadow: none !important;
  padding-left: 12px !important;
  padding-right: 12px !important;
}

.c11:hover {
  color: #fff;
  background: #003ef7;
  border-color: #003ef7;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px !important;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 24px !important;
  padding-bottom: 24px !important;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  margin-top: 84px;
}

@media (min-width:767px) {
  .c3 {
    grid-gap: 6px 6px;
    grid-template-columns: repeat( 7,1fr );
  }
}

@media (min-width:1024px) {
  .c3 {
    grid-template-columns: repeat( 8,1fr );
  }
}

@media (min-width:767px) {
  .c13 {
    grid-gap: 30px 30px;
    grid-template-columns: repeat( 3,1fr );
  }
}

@media (min-width:1024px) {
  .c13 {
    grid-template-columns: repeat( 3,1fr );
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2"
    >
      Transactions by stage
    </h2>
    <div
      className="c3"
    >
      <button
        className="c4"
        onClick={[Function]}
        selected={true}
      >
        All
      </button>
      <button
        className="c5"
        onClick={[Function]}
        selected={false}
      >
        Pre Ico
      </button>
      <button
        className="c5"
        onClick={[Function]}
        selected={false}
      >
        Ico
      </button>
      <button
        className="c5"
        onClick={[Function]}
        selected={false}
      >
        Final Ico
      </button>
    </div>
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <p
          className="c8"
        >
          0
           Transactions
        </p>
        <p
          className="c8"
        >
          Page 
          2
           / 
          0
        </p>
      </div>
      <div
        className="c9"
      >
        <select
          className="per-page"
          onChange={[Function]}
          value={30}
        >
          <option
            value=""
          >
            Per page
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="30"
          >
            30
          </option>
          <option
            value="60"
          >
            60
          </option>
          <option
            value="99"
          >
            99
          </option>
        </select>
        <button
          className="prev-page c10"
          onClick={[Function]}
        >
          Prev
        </button>
        <button
          className="next-page c11"
          onClick={[Function]}
        >
          Next
        </button>
      </div>
    </div>
    <hr
      className="c12"
    />
    <div
      className="transactions-list c13"
    />
  </div>
</div>
`;
